bugbug/tools/code_review.py

```diff
@@ -129,26 +286,87 @@ def get_review_request_by_id(self, revision_id: int) -> ReviewRequest:
         assert len(revisions) == 1
         return ReviewRequest(revisions[0]["fields"]["diffID"])
 
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(7),
+        wait=tenacity.wait_exponential(multiplier=1, min=16, max=64),
+        reraise=True,
+    )
     def get_patch_by_id(self, patch_id: int) -> Patch:
         assert phabricator.PHABRICATOR_API is not None
         raw_diff = phabricator.PHABRICATOR_API.load_raw_diff(int(patch_id))
         return Patch(raw_diff)
 
+    def _get_revisions_count(self):
+        return sum(1 for _ in phabricator.get_revisions())
+
+    def get_all_inline_comments(
+        self, comment_filter
+    ) -> Iterable[tuple[int, list[InlineComment]]]:
+        db.download(phabricator.REVISIONS_DB)
+
+        for revision in tqdm(
+            phabricator.get_revisions(), total=self._get_revisions_count()
+        ):
+            diff_comments: dict[int, list[InlineComment]] = defaultdict(list)
+
+            for transaction in revision["transactions"]:
+                if transaction["type"] != "inline":
+                    continue
+
+                # Ignore replies
+                if transaction["fields"]["replyToCommentPHID"] is not None:
+                    continue
+
+                # Ignore bot comments
+                if transaction["authorPHID"] == "PHID-USER-cje4weq32o3xyuegalpj":
+                    continue
+
+                if len(transaction["comments"]) != 1:
+                    print(transaction)
+                    # raise Exception("Unexpected, need to look into it")
+                    continue
+
+                transaction_comment = transaction["comments"][0]
+                comment_id = transaction_comment["id"]
+                date_created = transaction_comment["dateCreated"]
+                comment_content = transaction_comment["content"]["raw"]
+
+                diff_id = transaction["fields"]["diff"]["id"]
+                filename = transaction["fields"]["path"]
+                start_line = transaction["fields"]["line"]
+                end_line = (
+                    transaction["fields"]["line"] + transaction["fields"]["length"] - 1
+                )
+                # Unfortunately, we do not have this information for a limitation
+                # in Phabricator's API. We assume it as true as a workaround.
+                on_removed_code = True
```

Comment:
Why not using None to avoid any assumptions?

Reply:
I already added support for None in 70ae0d0490306872b635314a0f48839cd7d44ea2, but forgot to change it here. Fixed in fe2ff948144e76db67af2567519038b02f9b410d.